src/ripple/shamap/tests/SHAMap.test.cpp

```diff
@@ -173,11 +301,14 @@ class SHAMap_test : public beast::unit_test::suite
             keys[7].SetHex ("292891fe4ef6cee585fdc6fda1e09eb4d386363158ec3321b8123e5a772c6ca8");
 
             tests::TestFamily f{beast::Journal{}};
-            SHAMap map{SHAMapType::FREE, f};
+            SHAMap map{SHAMapType::FREE, f, v};
             if (! backed)
                 map.setUnbacked ();
             for (auto const& k : keys)
+            {
                 map.addItem(SHAMapItem{k, IntToVUC(0)}, true, false);
+                map.invariants();
+            }
```

Comment:
We can always use more testing.  But `invariants` is *very* thorough (and new with this P/R).  It not only checks whole-shaman invariants, it also crawls into every leaf and inner node, and checks each node's invariants too.  It is even checking design assumptions like every v2 inner node has at least two children.